### PR TITLE
Modify access right for cloud init users when oslogin is enabled/disabled

### PR DESCRIPTION
**Pull request by ya-mokin**

### CODE REVIEW

**Review by MaRVeLstas — APPROVED** [review]

File: google_guest_agent/cloud_init_accounts_test.go [hunk 1]
```diff
@@ -0,0 +1,150 @@
...
+
+package main
+
+import (
+	"github.com/GoogleCloudPlatform/guest-agent/metadata"
+	"path/filepath"
+	"testing"
+	"time"
+
+	"github.com/GoogleCloudPlatform/guest-agent/utils"
```
Comment (IlyaMar) [inline google_guest_agent/cloud_init_accounts_test.go:11]: Recommended to group imports: first system packages, then 3rd party, then ours.
https://github.com/golang/tools/commit/ed69e84b1518b5857a9f4e01d1f9cefdcc45246e

File: google_guest_agent/cloud_init_accounts_test.go [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+		}
+	}
+
+	actualResult, err := utils.ReadTextFile(sudoersFile)
+	if err != nil {
+		t.Errorf("Failed to read sudoers: %v", err)
+		return
+	}
+	if expectedResult != actualResult {
+		t.Fatalf(`Sudoers file is incorrect`)
```
Comment (IlyaMar) [inline google_guest_agent/cloud_init_accounts_test.go:148]: I reccomend logging actual and desired object, e g with assert.Equal()
https://pkg.go.dev/github.com/stretchr/testify/assert

File: metadata/user_data.go [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+		for _, v := range rules {
+			if rule, ok := v.(string); ok {
+				sudoRules = append(sudoRules, rule)
+			} else {
+				logger.Errorf("Cannot parse cloud-init sudo rule for user %s.", tempUser.Name)
+			}
+		}
+	} else if _, ok := tempUser.SudoRules.(bool); !ok {
+		logger.Errorf("Cannot parse sudo rules")
+	}
```
Comment (IlyaMar) [inline metadata/user_data.go:87]: tempUser.SudoRules.(bool) value not used, why do we convert it to bool?

Comment (ya-mokin) [reply metadata/user_data.go:87]: To check that yaml is valid. Possible types for this field are string, []string and bool

File: google_guest_agent/cloud_init_accounts_test.go (hunk 2 above)
Comment (ya-mokin) [reply google_guest_agent/cloud_init_accounts_test.go:148]: We're not going to add new framework at this time to keep changes to a minimum

**Review by IlyaMar — APPROVED** [review]

**Pull request merged**